src/components/Buttons/Button/Button.tsx

```diff
@@ -1,12 +1,32 @@
-import React, { forwardRef } from 'react';
-import PropTypes from 'prop-types';
+import React from 'react';
 import bem from '../../../utils/bem';
 import styles from './Button.scss';
-import { CONTEXTS, SIZES } from '../../../constants';
+import { ButtonTypes, ContextTypes, SizesTypes } from '../../../constants';
```

Comment:
I'd put `ButtonTypes, ContextTypes, SizesTypes` (and all the other types) in singular form.

Reply:
What about using `TButton`, `TContext`, etc.?

Reply:
I think we should stick with `Type` suffix (prefix?)
https://github.com/microsoft/TypeScript/blob/master/lib/typescript.d.ts#L2388

Do you guys follow any regularities in `Type` part moving around the variable name? Please help us if you do

Reply:
We started with prefixes but later decided not to use any prefixes and suffixes – it's too much noise and seems like it doesn't add much value.

Reply:
In this particular case `ButtonType` makes sense cause it's actually about a type of a button. But `ContextType` and `SizesType` are redundant and can be written as `Size` and `Context`. Leading capital indicates that it's a type or enum.

Reply:
Did you never have a problem with name clashes? e.g. there is a variable size, but also the var that describes it called the same? I guess more relevant for classes and such that are PascalCased

Reply:
That was my initial concern (conflicts of names) when we decided to use prefixes, but in real life for us that never happened. The only thing that I can imagine is `Class` thet implements some `Interface` with the same name. But again, never happend to us :)